oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/authorization/OAuth2AuthorizationServerConfigurer.java

```diff
@@ -314,6 +334,13 @@ private void initEndpointMatchers(ProviderSettings providerSettings) {
 				OAuth2AuthorizationServerMetadataEndpointFilter.DEFAULT_OAUTH2_AUTHORIZATION_SERVER_METADATA_ENDPOINT_URI, HttpMethod.GET.name());
 		this.oidcClientRegistrationEndpointMatcher = new AntPathRequestMatcher(
 				providerSettings.oidcClientRegistrationEndpoint(), HttpMethod.POST.name());
+		this.oidcUserInfoEndpointMatcher = new OrRequestMatcher(
+				new AntPathRequestMatcher(
+						OidcUserInfoEndpointFilter.DEFAULT_OIDC_USER_INFO_ENDPOINT_URI,
```

Comment:
Please add a new setting to `ProviderSettings` called `oidcUserInfoEndpoint` and use it here. It would default to `/userinfo` though `ProviderSettings` does not use constants from anywhere else in the codebase. Change the visibility of this constant in the filter to private.